src/bundle/ui-dev/src/modules/universal-discovery/components/tree-view/tree.view.js

```diff
@@ -0,0 +1,117 @@
+
+import ContentTreeModule from '../../../content-tree/content.tree.module';
+import { loadAccordionData } from '../../services/universal.discovery.service';
+import { getLocationData } from '../../content.meta.preview.module';
+import {
+    AllowedContentTypesContext,
+    ContainersOnlyContext,
+    ContentTypesMapContext,
+    LoadedLocationsMapContext,
+    MarkedLocationIdContext,
+    MultipleConfigContext,
+    RestInfoContext,
+    RootLocationIdContext,
+    SelectedLocationsContext,
+    SortOrderContext,
+    SortingContext,
+} from '../../universal.discovery.module';
+
+const flattenTree = (tree) => tree.reduce((output, branch) => [...output, branch.locationId, ...flattenTree(branch.subitems)], []);
+
+const TreeView = () => {
+    const [loadedLocationsMap, dispatchLoadedLocationsAction] = useContext(LoadedLocationsMapContext);
+    const [markedLocationId, setMarkedLocationId] = useContext(MarkedLocationIdContext);
+    const [multiple, multipleItemsLimit] = useContext(MultipleConfigContext);
+    const [selectedLocations, dispatchSelectedLocationsAction] = useContext(SelectedLocationsContext);
+    const [sortOrder, setSortOrder] = useContext(SortOrderContext);
+    const [sorting, setSorting] = useContext(SortingContext);
+    const allowedContentTypes = useContext(AllowedContentTypesContext);
+    const containersOnly = useContext(ContainersOnlyContext);
+    const contentTypesMap = useContext(ContentTypesMapContext);
+    const restInfo = useContext(RestInfoContext);
+    const rootLocationId = useContext(RootLocationIdContext);
+    const locationData = useMemo(() => getLocationData(loadedLocationsMap, markedLocationId), [markedLocationId, loadedLocationsMap]);
+    const expandItem = (item, event) => {
+        event.preventDefault();
+        event.currentTarget
+            .closest('.c-list-item__label')
+            .querySelector('.c-list-item__toggler')
+            .click();
+    };
+    const markLocation = (item) => {
+        const { locationId } = item;
+
+        if (locationId === markedLocationId) {
+            return;
+        }
+
+        loadAccordionData(
+            {
+                ...restInfo,
+                parentLocationId: locationId,
+                sortClause: sorting,
+                sortOrder: sortOrder,
+                rootLocationId,
+            },
+            (locationsMap) => {
+                const { location } = locationsMap[locationsMap.length - 1];
+                const contentTypeInfo = contentTypesMap[location.ContentInfo.Content.ContentType._href];
+                const isContainer = contentTypeInfo.isContainer;
+                const isNotSelectable =
+                    (containersOnly && !isContainer) || (allowedContentTypes && !allowedContentTypes.includes(contentTypeInfo.identifier));
+
+                setMarkedLocationId(locationId);
+                dispatchLoadedLocationsAction({ type: 'CUT_LOCATIONS', locationId: markedLocationId });
```

Comment:
Why do you `CUT` and in next line you do `SET` which replace the whole `loadedLocations`?